govcd/ip_space.go

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2023 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+// IpSpace provides structured approach to allocating public and private IP addresses by preventing
+// the use of overlapping IP addresses across organizations and organization VDCs.
+//
+// An IP space consists of a set of defined non-overlapping IP ranges and small CIDR blocks that are
+// reserved and used during the consumption aspect of the IP space life cycle. An IP space can be
+// either IPv4 or IPv6, but not both.
+//
+// Every IP space has an internal scope and an external scope. The internal scope of an IP space is
+// a list of CIDR notations that defines the exact span of IP addresses in which all ranges and
+// blocks must be contained in. The external scope defines the total span of IP addresses to which
+// the IP space has access, for example the internet or a WAN.
+type IpSpace struct {
+	IpSpace   *types.IpSpace
+	vcdClient *VCDClient
+}
+
+// CreateIpSpace creates IP Space with desired configuration
+func (vcdClient *VCDClient) CreateIpSpace(ipSpaceConfig *types.IpSpace) (*IpSpace, error) {
+	client := vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointIpSpaces
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	result := &IpSpace{
+		IpSpace:   &types.IpSpace{},
+		vcdClient: vcdClient,
+	}
+
+	err = client.OpenApiPostItem(apiVersion, urlRef, nil, ipSpaceConfig, result.IpSpace, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// GetAllIpSpaceSummaries retrieve summaries of all IP Spaces with an optional filter
+// Note. There is no API endpoint to get multiple IP Spaces with their full definitions. Only
+// "summaries" endpoint exists, but it does not include all fields. To retrieve complete structure
+// one can use `GetIpSpaceById` or `GetIpSpaceByName`
+func (vcdClient *VCDClient) GetAllIpSpaceSummaries(queryParameters url.Values) ([]*IpSpace, error) {
+	client := vcdClient.Client
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointIpSpaceSummaries
+	apiVersion, err := client.getOpenApiHighestElevatedVersion(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	typeResponses := []*types.IpSpace{{}}
+	err = client.OpenApiGetAllItems(apiVersion, urlRef, queryParameters, &typeResponses, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wrap all typeResponses into DefinedEntityType types with client
```

Comment:
This is probably leftover from other code

Reply:
Fixed.